pkgs/misc/emulators/citra/base.nix

```diff
@@ -0,0 +1,80 @@
+{ pname, version, src, branchDesc
+, lib, stdenv, config, fetchFromGitHub
+, catch2, cmake, ninja, pkg-config, wrapQtAppsHook
+, qtbase, qtdeclarative, qtmultimedia, qttools
+, boost17x, ffmpeg, libusb1, rapidjson, SDL2, zstd, libressl
+, alsaSupport      ? stdenv.isLinux, alsa-lib
+, pulseSupport     ? stdenv.isLinux, libpulseaudio
+, udevSupport      ? stdenv.isLinux, udev
+, sndioSupport     ? true, sndio
+, fdkSupport       ? true, fdk_aac
+, jackaudioSupport ? true, libjack2
+, discordSupport   ? true
+, onlineSupport    ? true
+}:
+
+let
+  inherit (lib) optional;
+  compat-src = ./compatibility_list.json; # Updated 1/12/22 from https://api.citra-emu.org/gamedb/
+in stdenv.mkDerivation {
+  inherit pname version src;
+
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=OFF"
+    "-DCITRA_ENABLE_COMPATIBILITY_REPORTING=ON"
+    "-DCITRA_USE_BUNDLED_FFMPEG=OFF"
+    "-DCITRA_USE_BUNDLED_QT=OFF"
+    "-DCITRA_USE_BUNDLED_SDL2=OFF"
+    "-DDYNARMIC_ENABLE_CPU_FEATURE_DETECTION=OFF"
```

Comment:
Is this necessary?
I gave a quick look at Dynarmic's source code, [it seems like](https://github.com/merryhime/dynarmic/blob/644172477eaf0d822178cb7e96c62b75caa96573/src/dynarmic/backend/x64/block_of_code.cpp#L78) CPU feature detection is done at runtime, not compile time.